### PR TITLE
Accessibility 101: first article of three

### PR DESCRIPTION
**Pull request by zersiax**
This is the first out of three articles I promised to write on the Curriculum development gitter channel.
This one deals with the very basics, more to come. I didn't really find a suitable branch to request to except for master . If this is the wrong branch, please advise on where you would rather see it merged.

### CODE REVIEW

**Comment by Rafase282** [comment]
Master is what you create your branch off. There are some things I will point out on the document.

File: accessibility-101.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+As I stated above, accessibility is important for a large group of different people, each with their own needs. Making your website work for literally everyone is a large, on-going task.
+
+To bring a bit of a method to the madness, the Web Content Accessibility Guidelines or [WCAG](https://www.wuhcag.com/web-content-accessibility-guidelines/) were composed. This document contains a number of criteria you can use to check your website.
+For now, I will cover some of the most important basics here. I will point you at the low-hanging fruits, so to speak. In subsequent articles, I will discuss more advanced techniques like [WAI-ARIA]  which is important for JavaScript-based apps.
+
+####Talk like the natives
+The HTML specification is a document that describes how the language should be used to build websites. Assistive technologies, like screenreaders, speech recognition programs etc. are aware of this document. Web developers however, often are not, or at least not enough, and think something like this is ok:
+
+```
```
Comment (Rafase282) [inline accessibility-101.md]: add html like this ```html

File: accessibility-101.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+ The second element looks like a heading visually after styling it with CSS, but semantically is a span. Therefore, assistive technologies won't know its a heading. A screenreader will read this as regular text, instead of a heading. Screenreaders often have a hotkey to quickly jump to the nearest heading, this heading will not be included in that scope.
+ 
+ The third element could for example be an element a user can click to change the language of the website. Maybe a fancy animated menu of languages will expand when it is clicked. However, this is also a span and does not expose its role (link, or button), making assistive technologies think this is just the word English with some styling.
+ 
+ Spans and divs are non-elements. They are meant to contain other elements, not to be elements themselves. You can fix these in two ways:
+ 
+- You can manually add ARIA-attributes to the elements above. This is an advanced topic and outside the scope of this article.
+- Or, you can simply do this:
+ 
+ ``
```
Comment (Rafase282) [inline accessibility-101.md]: You only have two back ticks instead of three. Also you should use html for syntax highlight.

File: accessibility-101.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+ 
+ ####A foundation cannot stand without structure
+ 
+ A bit earlier, I touched upon a screenreader's hotkeys to jump from heading to heading. There are in fact many hotkeys like this to quickly jump to the nearest table, form field, link etc.
+ Making sure these headings are actually in logical places is therefore a good practice and really decreases your assistive technology users'  stress levels, which is good if you want visitors to keep coming back to your website.
+ 
+ Also remember that headings are hierarchical.  If you use an h2, make sure the h3's that follow it actually have something to do with that h2. Don't put an h3 for contact details under your h2 for recent blog posts. 
+ A good analogy here is a book with chapters, that have subsections. You wouldn't put a section on baking cookies in the middle of a chapter on preparing vegetables ...or ...you wouldn't... right?
+ 
+####What's the alternative?
```
Comment (Rafase282) [inline accessibility-101.md]: Remember to add space after the #

File: accessibility-101.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+ I'm sure you would cringe if you had to fill in a form where light yellow, hopelessly loopy letters are placed on a white background. For people who's eyesight is not as good, like your grandma for example, this becomes hopelessly worse. 
+
+ The WCAG has contrast ratios for smaller and larger letters and there's plenty of tools out there to check if the contrast ratios are strong enough. The information and tooling is there, go use it :)
+ 
+####What does this button do?
+ While we are at the topic of forms, let's quickly glance at the <label> tag. This little guy is kinda important.
+ When you put some input fields on a web page, you can use labels to ...well ...label them. However, putting them next to each other is not quite enough. The attribute you want is the for-attribute, which takes the ID of a subsequent input field. This way, assistive technologies know what label to associate with what form field.
+ I guess the best way to illustrate this is by giving an example:
+ 
+```
```
Comment (Rafase282) [inline accessibility-101.md]: html

Comment (zersiax) [inline accessibility-101.md]: Hi,

Alright. So add html to the code blocks and a space to headings. Why
is the second one necessary btw? The headings render fine here in the
live preview.
Requesting a merge to the master branch was alright, then?

Comment (Rafase282) [inline accessibility-101.md]: Yes for the wiki you use master, for the FCC you use staging.

for the code blocks you can specify what type of code it is like css, html, javascript or js and so on.

As for the heading it might just be me then, My linter adds space to them and so I have gotten used to that as the standard, it also looks better when reading the markdown code.

However, notice the back ticks, there was a whole section with headings and such that was rendered as a quotation or something else. Could be because of that. I used the live preview here by the way.

Comment (zersiax) [inline accessibility-101.md]: I added the changes you suggested. Is there anything else, or shall I
make another pull request?

Comment (zersiax) [inline accessibility-101.md]: Ugh amateur hour last night, apparently :) I forgot subsequent commits
are added to the pull request automagically. anyway, I committed my
changes right before sending the previous email. I noticed through
some glitch that an older version of the article had been pushed, so
it should be a lot better now.

File: accessibility-101.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+ The second element looks like a heading visually after styling it with CSS, but semantically is a span. Therefore, assistive technologies won't know its a heading. A screenreader will read this as regular text, instead of a heading. Screenreaders often have a hotkey to quickly jump to the nearest heading, this heading will not be included in that scope.
+ 
+ The third element could for example be an element a user can click to change the language of the website. Maybe a fancy animated menu of languages will expand when it is clicked. However, this is also a span and does not expose its role (link, or button), making assistive technologies think this is just the word English with some styling.
+ 
+ Spans and divs are non-elements. They are meant to contain other elements, not to be elements themselves. You can fix these in two ways:
+ 
+- You can manually add ARIA-attributes to the elements above. This is an advanced topic and outside the scope of this article.
+- Or, you can simply do this:
+ 
+ ``html
```
Comment (Rafase282) [inline accessibility-101.md]: You forgot to add the third backtick. ```html

Comment (Rafase282) [inline accessibility-101.md]: You should fix the code block. ```html

Comment (zersiax) [inline accessibility-101.md]: Fixed :)

**Comment by Rafase282** [comment]
Good now just squash them together and I'll be happy to merge it https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one

**Comment by zersiax** [comment]
Sorry for the delay, I missed your last message. Rebase done, I think.
Let me know if anything else needs to be done :)

**Comment by Rafase282** [comment]
Good work @zersiax

File: accessibility-101.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
+## Accessibility basics: What is this thing you speak of?
+
+
+> "The Dark Arts are many, varied, ever-changing, and eternal. 
+> Fighting them is like fighting a many-headed monster, which, each time a neck is severed, sprouts a head even fiercer and cleverer than before. 
+> You are fighting that which is unfixed, mutating, indestructible."
+>
+> —Professor Severus Snape, Harry Poter Series
```
Comment (SaintPeter) [inline accessibility-101.md]: Missing `t` in Potter

File: accessibility-101.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+ <button>This is a button</button>
+ 
+ <h2>Here's a heading level two</h2>
+ 
+ <a href=" JavascriptThing">English</a>
+ ```
+ 
+ Boom. Suddenly, all these elements are now perfectly accessible, just by using native HTML. HTML the way it was meant to be used, in other words.
+ 
+ #### A foundation cannot stand without structure
```
Comment (SaintPeter) [inline accessibility-101.md]: This section is unintentionally indented?

File: accessibility-101.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+As I stated above, accessibility is important for a large group of different people, each with their own needs. Making your website work for literally everyone is a large, on-going task.
+
+To bring a bit of a method to the madness, the Web Content Accessibility Guidelines or [WCAG](https://www.wuhcag.com/web-content-accessibility-guidelines/) were composed. This document contains a number of criteria you can use to check your website.
+For now, I will cover some of the most important basics here. I will point you at the low-hanging fruits, so to speak. In subsequent articles, I will discuss more advanced techniques like [WAI-ARIA]  which is important for JavaScript-based apps.
+
+#### Talk like the natives
+The HTML specification is a document that describes how the language should be used to build websites. Assistive technologies, like screenreaders, speech recognition programs etc. are aware of this document. Web developers however, often are not, or at least not enough, and think something like this is ok:
+
+```html
+<div class="awesome-button"</div>
```
Comment (SaintPeter) [inline accessibility-101.md]: Missing closing `>`.
Should be:
`<div class="awesome-button></div>`

Comment (zersiax) [inline accessibility-101.md]: Hmm. Good catches, but the page is live now. Would I have to make a
new branch, fix these issues and submit another PR, or is there a
quicker way to fix hese?

Comment (SaintPeter) [inline accessibility-101.md]: Yes, a new Branch and new PR would be the way to go to roll up the fixes.

Comment (zersiax) [inline accessibility-101.md]: Very well. Please let me know any others you find, so I can lump them
all together in one PR. Also, please explain what mistake you found in
that last comment, you pointed at a bit of text but didn't specify the
error for as far as I can see.

File: accessibility-101.md (hunk 7 above)
Comment (SaintPeter) [inline accessibility-101.md]: image

**Comment by SaintPeter** [comment]
Those are all the errors I found.

**Comment by zersiax** [comment]
At the risk of sounding stupid ...what is that image, a screenshot?
You are aware I am fully blind? :)

**Comment by SaintPeter** [comment]
Well, now I feel like an idiot.  No, I was not aware.  Yes it was a screenshot.

That paragraph, starting with "#### A foundation cannot stand without structure" is indented.  I can't tell why, maybe a leading space or something is keeping it at the same indent level as the bullet point before it.

**Comment by zersiax** [comment]
Hahaha :) No problem,  it happens. :) And I know why that happened,
notepad++'s auto-indent sucks rather badly. I forgot my laptop charger
at home for the weekend so I have to use my battery sparingly, but
i'll make sure to fix these cosmetic issues on sunday. :)
